src/javascript/DesignSystem/DatePicker/DatePicker.jsx

```diff
@@ -39,37 +39,83 @@ const DatePickerCmp = ({
     classes,
     onSelectDateTime,
     selectedDateTime,
+    disabledDays: incomingDisableDays,
     ...props
 }) => {
     const [selectedDays, setSelectedDays] = useState([]);
-    const [selectedHour, setSelectedHour] = useState('0:00');
+    const [selectedHour, setSelectedHour] = useState('00:00');
+    const isDateTime = variant === 'datetime';
 
     const locale = locales[lang] || {};
 
     const selectedDate = selectedDateTime ? [selectedDateTime] : selectedDays;
-    const isDateTime = variant === 'datetime';
+    let disabledHours = {};
+    const disabledDays = [];
+    const extractDateAndHours = (entry, boundIncludeOffset) => {
+        return {
+            date: !isDateTime && entry.include ? entry.date : undefined,
+            hours: isDateTime && dayjs(entry.date).format('YYYYMMDD') === dayjs(selectedDate).format('YYYYMMDD') ? dayjs(entry.date).add(entry.include ? boundIncludeOffset : 0, 'minute').format('HH:mm') : undefined
+        };
+    };
+
+    const disabled = {};
+    incomingDisableDays.forEach(entry => {
```

Comment:
Could be great to externalize this code in a fct to get the disabledDays and disabledHours like the fct extractDateAndHours.
it could return: {disabledDays: [], disabledHours: {before:..., after...}}